src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -175,11 +176,15 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
                                     transactionId
                             );
                             return !amountTotal.equals(requestAuthorizationRequestDto.getAmount())
-                                    ? Mono.empty()
+                                    ? Mono.error(
+                                            new TransactionAmountMismatchException(
+                                                    requestAuthorizationRequestDto.getAmount(),
+                                                    amountTotal
+                                            )
+                                    )
```

Comment:
Should we treat this error case as a transactions not found exception or a bad request?

Reply:
If we are to differentiate the two cases I'd vote for 422 and reserve 400 for purely formal errors in the request body

Reply:
bikeshedding time come back :)  
I'd vote for a generic 400 bad request, just to avoid to give more then the required information to the frontend

Reply:
Restored to generic bad request return code, shall we go?

Reply:
Voted for 409

Reply:
sounds good